File: snippets/fsharp/System/Guid/Parse/parseex1.fs
module parseex1

// <Snippet3>
open System

let originalGuid = Guid.NewGuid()
// Create an array of string representations of the GUID.
let stringGuids =
    [| originalGuid.ToString "B"
       originalGuid.ToString "D"
       originalGuid.ToString "X" |]

// Parse each string representation.
for stringGuid in stringGuids do
    try
        let newGuid = Guid.Parse stringGuid
        printfn $"Converted {stringGuid} to a Guid"
    with
    | :? ArgumentNullException ->
        printfn "The string to be parsed is null."
    | :? FormatException ->
        printfn $"Bad format: {stringGuid}"

// The example displays output similar to the following:
//
//    Converted {81a130d2-502f-4cf1-a376-63edeb000e9f} to a Guid
//    Converted 81a130d2-502f-4cf1-a376-63edeb000e9f to a Guid
//    Converted {0x81a130d2,0x502f,0x4cf1,{0xa3,0x76,0x63,0xed,0xeb,0x00,0x0e,0x9f}} to a Guid
// </Snippet3>

File: snippets/csharp/System/SByte/Parse/parseex1.cs
using System;

public class Example
{
    public static void Main()
    {
        // <Snippet1>
        // Define an array of numeric strings.
        string[] values = { "-16", "  -3", "+ 12", " +12 ", "  12  ",
                          "+120", "(103)", "192", "-160" };

        // Parse each string and display the result.
        foreach (string value in values)
        {
            try
            {
                Console.WriteLine($"Converted '{value}' to the SByte value {sbyte.Parse(value)}.");
            }
            catch (FormatException)
            {
                Console.WriteLine($"'{value}' cannot be parsed successfully by SByte type.");
            }
            catch (OverflowException)
            {
                Console.WriteLine($"'{value}' is out of range of the SByte type.");
            }
        }
        // The example displays the following output:
        //       Converted '-16' to the SByte value -16.
        //       Converted '  -3' to the SByte value -3.
        //       '+ 12' cannot be parsed successfully by SByte type.
        //       Converted ' +12 ' to the SByte value 12.
        //       Converted '  12  ' to the SByte value 12.
        //       Converted '+120' to the SByte value 120.
        //       '(103)' cannot be parsed successfully by SByte type.
        //       '192' is out of range of the SByte type.
        //       '-160' is out of range of the SByte type.
        // </Snippet1>
    }
}
